### PR TITLE
Add more info for creating volume snapshots

### PR DESCRIPTION
**Pull request by anuragaw**
Add more details in console logs and event details

Also show popup on successful completition of snapshots

## Description

Fixes: #3327
## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## Screenshots (if appropriate):
Screenshot from 2019-06-09 19-29-00

### CODE REVIEW

File: ui/scripts/storage.js [hunk 1]
```diff
@@ -944,7 +944,14 @@
...
                                                 _custom: {
-                                                    jobId: jid //take snapshot from a volume doesn't change any property in this volume. So, don't need to specify getUpdatedItem() to return updated volume. Besides, createSnapshot API doesn't return updated volume.
+                                                    jobId: jid, //take snapshot from a volume doesn't change any property in this volume. So, don't need to specify getUpdatedItem() to return updated volume. Besides, createSnapshot API doesn't return updated volume.
+                                                    onComplete: function(json, customData) {
+                                                        var volumeId = json.queryasyncjobresultresponse.jobresult.snapshot.volumeid;
+                                                        var snapshotId = json.queryasyncjobresultresponse.jobresult.snapshot.id;
+                                                        cloudStack.dialog.notice({
+                                                            message: "Created snapshot for volume " + volumeId + " with snapshot id " + snapshotId
```
Comment (rhtyd) [inline ui/scripts/storage.js]: nit - `id` should be written as `ID` wherever possible.

**Comment by shwstppr** [comment]
@anuragaw Original issue asks for volume name in logs. Should that be shown? Although that might be inconsistent with default policy where the code should only be showing UUIDs.

**Comment by rhtyd** [comment]
@anuragaw left minor nit, otherwise LGTM.

**Comment by anuragaw** [comment]
> @anuragaw Original issue asks for volume name in logs. Should that be shown? Although that might be inconsistent with default policy where the code should only be showing UUIDs.

The issue asks for volume name but uses UUID in examples. Also I think given names may not be unique it's more helpful for logs to contain UUID. Any input to suggest otherwise is welcome.

File: ui/scripts/storage.js (hunk 1 above)
Comment (anuragaw) [reply ui/scripts/storage.js]: changing @rhtyd

**Comment by rhtyd** [comment]
@shwstppr can you test and review this?

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by anuragaw** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@anuragaw @shwstppr check build failure using Travis's job 1 or check backend Jenkin's console. I can see 
```
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.024 s <<< FAILURE! - in org.apache.cloudstack.api.command.test.CreateSnapshotCmdTest
[ERROR] testCreateFailure(org.apache.cloudstack.api.command.test.CreateSnapshotCmdTest)  Time elapsed: 0.006 s  <<< ERROR!
java.lang.NullPointerException
	at org.apache.cloudstack.api.command.test.CreateSnapshotCmdTest.testCreateFailure(CreateSnapshotCmdTest.java:127)
[ERROR] Errors: 
[ERROR]   CreateSnapshotCmdTest.testCreateFailure:127 » NullPointer
[ERROR] Tests run: 111, Failures: 0, Errors: 1, Skipped: 0
```

**Comment by anuragaw** [comment]
This seems unrelated. Will investgate and get back @rhtyd @shwstppr

**Comment by anuragaw** [comment]
Rebased and also fixed the errror of failing test.

**Comment by anuragaw** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by ustcweizhou** [comment]
code LGTM, did not test it.

@anuragaw could you squash the commits ?

**Comment by anuragaw** [comment]
@ustcweizhou - can we directly do a squash and merge into master? Seems redundant given all PRs are merged into master after a squash. Can do it though if you think it would any value here too.

**Pull request merged**